/**
 * File         : AngkaSial.java 29/03/2023
 * Penulis      : Yesy Margharetta Munthe
 * Deskripsi  :Program penggunaan exception buatan sendiri
               Pengenalan klausa 'throw' dan ' throws'
 */
public class AngkaSial{
	
	public void cobaAngka(int angka) throws AngkaSialException{
		if(angka==13) {
			throw new AngkaSialException();
		}
		System.out.println(angka+" bukan angka sial");
	}
	
	public static void main(String [] argas){
		AngkaSial as = new AngkaSial();
		try{
			as.cobaAngka(10);
			as.cobaAngka(13);
			as.cobaAngka(12);
		}catch(AngkaSialException ase){
			//method getMessage() telah ada pada kelas "Exception:
			System.out.println(ase.getMessage());
			System.out.println("hati-hati memasukkan angka!1!");
		}
	}
}
// Jawaban Pertanyaan
// Baris ke 12  dan 21 tidak bisa dieksekusi karena class angkaSial exception belum dibuat
// karena itu diperlukan pembuatan class angka sial exception yang merupakan turunan dari class exception